Documentation/ApiOverview/Database/ConnectionPool/Index.rst

```diff
@@ -1,56 +1,57 @@
-
-
-The `QueryBuilder` is the default workhorse object used by extension authors to express complex queries,
-while a `Connection` instance can be used as shortcut to deal with some simple query cases and little written down code.
+..  literalinclude:: _MyTableRepository.php
+    :caption: EXT:my_extension/Classes/Domain/Model/MyTableRepository.php
 
+The :php:`QueryBuilder` is the default workhorse object used by extension
+authors to express complex queries, while a :php:`Connection` instance can be
+used as a shortcut to handle some simple query cases and little written code.
 
 Pooling
 =======
 
-TYPO3 can handle multiple connections to different database endpoints at the same time. This
-can be configured on a per-table basis in :php:`$GLOBALS['TYPO3_CONF_VARS']`. It allows running tables
-on different databases, without an extension developer taking care of that.
-
-The `ConnectionPool` implements this feature: It looks up a configured table-to-database
-mapping and can return a `Connection` or a `QueryBuilder` instance for that specific connection.
-Those objects internally know which target connection they are dealing with and will
-for instance quote field names accordingly.
-
-The transparency of tables to different database endpoints is limited, though:
-
-Executing a table `JOIN` between two tables that point to different connections will throw an exception.
-This restriction may in practice create implicit "groups" of tables that need to point to one connection
-at once if an extension or the TYPO3 Core  joins those tables.
-
-This can turn out as a headache if multiple different extensions use for instance the Core category or
-collection API with their mm table joins between Core internal tables and their extension's counterparts.
-
-That situation is not easy to deal with. At the time of this writing the Core development will
-eventually implement some non-join fallbacks for typical cases that would be good to decouple, though.
-
-.. tip::
-
-   In case joins cannot be decoupled but still affected tables must run on different databases,
-   and if the code can not be easily adapted, some DBMS like `PostgreSQL` allow executing those
-   queries by having own connection handlers to different other endpoints on its own.
+TYPO3 can handle multiple connections to different database endpoints at the
+same time. This can be configured for each individual table in
+:php:`$GLOBALS['TYPO3_CONF_VARS']` (see :ref:`database configuration
+<database-configuration>` for details). This makes it possible to run tables on
+different databases without an extension developer having to worry about it.
+
+The :php:`ConnectionPool` implements this feature: It looks for configured
+table-to-database mapping and can return a :php:`Connection` or a
+:php:`QueryBuilder` instance for that specific connection. These objects know
+internally which target connection they are dealing with and will quote field
+names accordingly, for instance.
+
+Caveat
```

Comment:
Where does this come from? Did you add the headline? What does it mean?

Reply:
Yup, it's from me: "Caveat - a warning that particular things need to be considered before something can be done"
https://www.oxfordlearnersdictionaries.com/definition/english/caveat

Reply:
Maybe the formulation is a little over the top as we are adressing ppl from all over the world, not just native speakers?

Reply:
Okay, then "Beware" ;-)
I haven't seen this word quite some time. But I am open for suggestions ;-)

Reply:
Beaware is good